Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -2385,6 +2387,7 @@ AODProducerWorkflowDPL::TrackExtraInfo AODProducerWorkflowDPL::processBarrelTrac
       bcOfTimeRef = fillBCSlice(extraInfoHolder.bcSlice, t - error, t + error, bcsMap);
     }
     extraInfoHolder.trackTime = float(t - bcOfTimeRef * o2::constants::lhc::LHCBunchSpacingNS);
+    extraInfoHolder.diffBCRef = int(bcOfTimeRef - mStartIR.toLong());
```

Comment:
I don't see where you use this. The point was to memorize the reference of the track time so that in the processBarrelTrackQA you can save time0 wrt stored track time. 
If done correctly, the delta time0 you will store should not exceed TPC drift time.

Reply:
I utilized the memorized value in the code snippet below. In the code where you suggested using it, I don't have access to extraInfoHolder.diffBCRef. Therefore, I placed it elsewhere.

https://github.com/AliceO2Group/AliceO2/blob/06752a4956196ee918ad5aa1daad20f892b5ae99/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L478-L479

Reply:
Indeed, did not notice this line, sorry. Could you dump values at https://github.com/AliceO2Group/AliceO2/pull/12387/commits/06752a4956196ee918ad5aa1daad20f892b5ae99#diff-6dc278732c5ed5f43d5567ad2c469ac11acf0e85339ae708bdf977ddd8fba392R477 the values, i.e.
```
LOGP(info, "orig time0 in bc: {} diffBCRef: {}, ttime: {} -> {}", trackQAInfoHolder.tpcTime0*8, extraInfoHolder.diffBCRef, extraInfoHolder.trackTime, (trackQAInfoHolder.tpcTime0 * 8 - extraInfoHolder.diffBCRef) * o2::constants::lhc::LHCBunchSpacingNS - extraInfoHolder.trackTime);
```

Reply:
I see infinity, I will check in the debugger which value is wrong ...
```
LOGP(info, "orig time0 in bc: {} diffBCRef: {}, ttime: {} -> {}", trackQAInfoHolder.tpcTime0*8, extraInfoHolder.diffBCRef, extraInfoHolder.trackTime, (trackQAInfoHolder.tpcTime0 * 8 - extraInfoHolder.diffBCRef) * o2::constants::lhc::LHCBunchSpacingNS - extraInfoHolder.trackTime);
```

```
[43011:aod-producer-workflow]: [17:06:09][INFO] orig time0 in bc: inf diffBCRef: -1443076264, ttime: 4565.9937 -> inf
[43011:aod-producer-workflow]: [17:06:09][INFO] orig time0 in bc: inf diffBCRef: -1443076264, ttime: 4565.9937 -> inf
[43011:aod-producer-workflow]: [17:06:09][INFO] orig time0 in bc: inf diffBCRef: -1443076264, ttime: 4565.9937 -> inf
[43011:aod-producer-workflow]: [17:06:09][INFO] orig time0 in bc: inf diffBCRef: -1443076264, ttime: 4565.9937 -> inf
```

Reply:
sorry, actually the https://github.com/AliceO2Group/AliceO2/blob/06752a4956196ee918ad5aa1daad20f892b5ae99/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L2390 should be just 
```
extraInfoHolder.diffBCRef = int(bcOfTimeRef);
```
I guess you put this printout after the line 
```
trackQAInfoHolder.tpcTime0 = (trackQAInfoHolder.tpcTime0 * 8 - extraInfoHolder.diffBCRef) * o2::constants::lhc::LHCBunchSpacingNS - extraInfoHolder.trackTime;
```
? I meant to put it before, so that 1st the original time0 is printed (it cannot be inf).

Reply:
Fixed https://github.com/AliceO2Group/AliceO2/pull/12387/commits/8e00e22372e56915ffe7c6d315dcf80cb1165cbc

Distribution as expected:
<img width="812" alt="image" src="https://github.com/AliceO2Group/AliceO2/assets/21962765/9ecca3c3-8f10-4783-a3a1-bcfd8f859dd1">